saleor/order/emails.py

```diff
@@ -5,52 +5,78 @@
 from templated_email import send_templated_mail
 
 from ..core.utils import build_absolute_uri
+from .emails_helpers import get_order_confirmation_markup
 from .models import Order
 
 CONFIRM_ORDER_TEMPLATE = 'source/order/confirm_order'
 CONFIRM_PAYMENT_TEMPLATE = 'source/order/payment/confirm_payment'
 CONFIRM_NOTE_TEMPLATE = 'source/order/note/confirm_note'
 
 
-def _send_confirmation(email, url, template, context=None):
+def get_email_context(order_token):
+    """Prepares context required for email template rendering."""
     site = Site.objects.get_current()
+    order_url = build_absolute_uri(
+        reverse('order:details', kwargs={'token': order_token}))
     ctx = {
         'protocol': 'https' if settings.ENABLE_SSL else 'http',
         'site_name': site.name,
         'domain': site.domain,
-        'url': url}
-    if context:
-        ctx.update(context)
-    send_templated_mail(
-        from_email=settings.ORDER_FROM_EMAIL,
-        recipient_list=[email],
-        context=ctx,
-        template_name=template)
+        'url': order_url}
+    return ctx
 
 
 def collect_data_for_email(order_pk, template):
+    """Collects data required for email sending.
+
+    Args:
+        order_pk (int): order primary key
+        template (str): email template path
+    """
     order = Order.objects.get(pk=order_pk)
-    email = order.get_user_current_email()
-    url = build_absolute_uri(
-        reverse('order:details', kwargs={'token': order.token}))
-    return {'email': email, 'url': url, 'template': template}
+    recipient_email = order.get_user_current_email()
+    email_context = get_email_context(order.token)
+
+    # Order confirmation template requires additional information
+    if template == CONFIRM_ORDER_TEMPLATE:
+        email_markup = get_order_confirmation_markup(order)
+        email_context.update({'order': order, 'schema_markup': email_markup})
+    return {
+        'recipient_email': recipient_email, 'template': template,
+        'context': email_context}
+
+
+def _send_confirmation(recipient_email, template, context):
```

Comment:
What's the benefit of using this function? Can't we just use `send_templated_email` directly in `send_order_confirmation` and other functions that send emails below?